### ISSUE TITLE
ref, complexType in element-tag, documentation in non-leaf-tags

### ISSUE THREAD

**Issue by PiQ99** [fnogatz/xsd2json#8]
Hey,

at first I have to say, I like your work. It is a great tool that helps me a lot at the moment. In addition I wanted to ask, if you plan further development? There are several things to do I guess.
Of course I saw, you did not commit anything about two months now, but maybe you continued developing at home? 

e.g. what about:
- ref-attributes in element-tags
- the documentation-attibute in a complexType (it gets ignored I guess atm?)
- complexTypes in element-tags?

These are several things that do not work atm, afaik, and I wanted to ask, if you are going to fix these things, maybe? 

Thanks for your answer and have a nice week!

**Comment by fnogatz** [comment]
Yes, the aim is to extend xsd2json step by step. Currently I'm working on the integration of `xs:import` and `xs:include` with its multiple schema support.

The best way to get things fixed is to provide some concrete examples. Feel free to open a pull request containing only some test cases in the `/test/xsd` and `/test/json` directories. The `ref` support should not be that hard, `xs:documentation` and `xs:annotation` too. What do you mean by your last bullet point?

**Comment by fnogatz** [comment]
Current status:
- [ ] ref-attributes in element-tags
- [x] the documentation-attibute in a complexType
- [ ] complexTypes in element-tags

**Comment by stevenroose** [comment]
@fnogatz I have an `xs:complexType` that does not appear in the JSON schema. Does that fall under this issue? It looks like this:
```xml
    <xs:complexType name="DocumentLineType1Choice">
        <xs:choice>
            <xs:element name="Cd" type="ExternalDocumentLineType1Code"/>
            <xs:element name="Prtry" type="Max35Text"/>
        </xs:choice>
    </xs:complexType>
```
